pkg/controller/clusterinstallation/minio.go

```diff
@@ -28,6 +28,28 @@ func (r *ReconcileClusterInstallation) checkMinio(mattermost *mattermostv1alpha1
 }
 
 func (r *ReconcileClusterInstallation) checkMinioSecret(mattermost *mattermostv1alpha1.ClusterInstallation, reqLogger logr.Logger) error {
+	// Check if custom secret was specified
+	if mattermost.Spec.Minio.Secret != "" {
+		// Check if the Secret exists
+		var secret *corev1.Secret
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: mattermost.Spec.Minio.Secret, Namespace: mattermost.Namespace}, secret)
+		if err != nil {
+			// Secret do not exist
+			return err
+		}
+
+		// Check if the Secret has required fields
+		if _, ok := secret.Data["accesskey"]; !ok {
+			return err
```

Comment:
Returning `err` here doesn't actually contain information regarding if "accesskey" is set. We will probably want something like `return fmt.Errorf("custom minio secret %s does not have an 'accesskey' value", mattermost.Spec.Minio.Secret)`

Reply:
Yes. I agree, since I already added a check this return will always return `nil` basically. I'll fix it. Thanks for pointing it out.